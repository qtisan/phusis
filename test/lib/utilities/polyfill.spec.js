import '../../..';
import { caught } from '../../..';

describe('Would these be the build-in functions?', () => {
  it('Date.prototype.getStamp()', () => {
    const d = new Date('2008-8-8');
    expect(d.getStamp()).toEqual(1218124800);
  });

  it('String.prototype.isCnNewID()', () => {
    const str = '189473',
      newid = '330382198608110018';
    expect(str.isCnNewID()).not.toBeTruthy();
    expect(newid.isCnNewID()).toBeTruthy();
  });

  it('caught()', () => {
    const error = new Error('I am an error!');
    const newError = caught('I hide the real error!');
    const newEntireError = caught(error, 'I DID hide the real error!', 109);
    expect(newError instanceof Error).toBeTruthy();
    expect(newEntireError).toMatchSnapshot();
    // newEntireError.log();
  });
});
