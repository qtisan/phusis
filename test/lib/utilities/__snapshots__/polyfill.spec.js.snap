// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Would these be the build-in functions? caught() 1`] = `[Error: I DID hide the real error!]`;
